bugsnag-maze-runner.gemspec

```diff
@@ -7,7 +7,7 @@ Gem::Specification.new do |spec|
   spec.version = BugsnagMazeRunner::VERSION
   spec.authors = ['Steve Kirkland']
   spec.email   = ['steve@bugsnag.com']
-  spec.required_ruby_version = '>= 2.3.0'
+  spec.required_ruby_version = '>= 2.5.0'
```

Comment:
It's the `rescue` in a `do`/`end` that was added in 2.5, as described in https://blog.bigbinary.com/2017/10/24/ruby-2.5-allows-rescue-inside-do-end-blocks.html.  (Which I only know because Rubocop pointed it out to me).

I think we're likely to add more `rescue`s as we build more resilience into the system, so it would be good to step up to 2.5 to benefit from the more succinct form.  Unless of course there are reasons for keeping to 2.3 specifically?